pallets/loans/src/lib.rs

```diff
@@ -916,16 +914,16 @@ impl<T: Config> Pallet<T> {
     fn redeem_allowed(
         currency_id: &CurrencyId,
         redeemer: &T::AccountId,
-        redeem_amount: Balance,
-    ) -> DispatchResult {
+        voucher_amount: Balance,
+    ) -> Result<Balance, DispatchError> {
```

Comment:
It is odd to return a balance here. I suggest putting this check into `redeem` and `redeem_all` separately.

Reply:
I think it's ok because `ensure_origin` also returns back the AccountId:

> I suggest putting this check into redeem and redeem_all separately.

Reply:
To be consistent with other methods, such as Borrow_Allowed, there may be liquidate_allowed. This is just to see if you can continue.

Reply:
do you have an idea on how should I write it?